packages/data/src/world/mq/gerudo_training_grounds_mq.yml

```diff
@@ -23,40 +19,171 @@
+  locations:
+    "MQ Gerudo Training Grounds SR Lava Back-Left": "event(GTG_MQ_LAVA_TORCH_COMMON_PLATFORMS_WATER_SIDE)"
+    "MQ Gerudo Training Grounds SR Lava Back-Right": "event(GTG_MQ_LAVA_TORCH_COMMON_PLATFORMS_WATER_SIDE)"
+    "MQ Gerudo Training Grounds SR Lava Center": "event(GTG_MQ_LAVA_TORCH_COMMON_PLATFORMS_WATER_SIDE)"
+    "MQ Gerudo Training Grounds SR Lava Front-Right": "event(GTG_MQ_LAVA_TORCH_COMMON_PLATFORMS_WATER_SIDE)"
+    "MQ Gerudo Training Grounds SR Lava Front-Left": "event(GTG_MQ_LAVA_TORCH_COMMON_PLATFORMS_WATER_SIDE)"
+    "MQ Gerudo Training Grounds SR Lava Front": "event(GTG_MQ_LAVA_TORCH_COMMON_PLATFORMS_WATER_SIDE)"
+"Gerudo Training Grounds Lava Room Maze Ledge":
+  dungeon: GTG
+  exits:
+    "Gerudo Training Grounds Lava Room Near Beginning": "can_hookshot || climb_anywhere || event(GTG_MQ_LAVA_TORCH_FAR) || (trick(OOT_MQ_GTG_FLAMES) && has_hover_boots)"
+    "Gerudo Training Grounds Lava Room Ledge Before Water Room": "can_longshot || hookshot_anywhere || event(GTG_MQ_LAVA_TORCH_FAR) || (can_hookshot && event(GTG_LAVA_HOOK_TARGETS)) || (has_hover_boots && event(GTG_LAVA_HOOK_TARGETS) && trick(OOT_MQ_GTG_FLAMES))"
+    "Gerudo Training Grounds Lava Room Deep Side Ledge": "hookshot_anywhere || climb_anywhere"
+    "Gerudo Training Grounds Maze Right Room 2": "true"
+  events:
+    GTG_LAVA_HOOK_TARGETS: "silver_rupees_gtg_lava_mq"
+    GTG_MQ_LAVA_TORCH_FAR: "has_fire_arrows"
+    GTG_MQ_LAVA_TORCH_COMMON_PLATFORMS_FAR: "(event(GTG_MQ_LAVA_TORCH_FAR) && (is_adult || (is_child && (has_hover_boots || climb_anywhere || ((event(GTG_LAVA_HOOK_TARGETS) && can_hookshot) || hookshot_anywhere))))) || (trick(OOT_MQ_GTG_FLAMES) && (has_hover_boots || (can_hookshot && event(GTG_LAVA_HOOK_TARGETS)) || hookshot_anywhere))"
+    #GTG_MQ_LAVA_SR_HOOK_ANYWHERE: "settings.silverRupeeShuffle && hookshot_anywhere"
+  locations:
+    "MQ Gerudo Training Grounds SR Lava Back-Left": "event(GTG_MQ_LAVA_TORCH_COMMON_PLATFORMS_FAR)"
+    "MQ Gerudo Training Grounds SR Lava Back-Right": "event(GTG_MQ_LAVA_TORCH_COMMON_PLATFORMS_FAR)"
+    "MQ Gerudo Training Grounds SR Lava Center": "event(GTG_MQ_LAVA_TORCH_COMMON_PLATFORMS_FAR)"
+    "MQ Gerudo Training Grounds SR Lava Front-Right": "event(GTG_MQ_LAVA_TORCH_COMMON_PLATFORMS_FAR)"
+    "MQ Gerudo Training Grounds SR Lava Front-Left": "event(GTG_MQ_LAVA_TORCH_COMMON_PLATFORMS_FAR)"
+    "MQ Gerudo Training Grounds SR Lava Front": "event(GTG_MQ_LAVA_TORCH_COMMON_PLATFORMS_FAR)"
+"Gerudo Training Grounds Lava Room Deep Side Ledge":
+  dungeon: GTG
+  exits:
+    "Gerudo Training Grounds Lava Room Near Beginning": "hookshot_anywhere || (event(GTG_MQ_LAVA_TORCH_FAR) && (is_adult || (is_child && (climb_anywhere || has_hover_boots))))"
+    "Gerudo Training Grounds Lava Room Maze Ledge": "(event(GTG_MQ_LAVA_HAMMER_SWITCH) && can_longshot) || (event(GTG_LAVA_HOOK_TARGETS) && (is_adult || (is_child && has_hover_boots)) && can_hookshot) || hookshot_anywhere || climb_anywhere"
```

Comment:
Hammer switch should be Hook Targets (I also dont know why you specified the hook part in the event name)

Reply:
I added the event and the trick, and removed the age part here.